### PR TITLE
SUBMARINE-316. Support listen k8s event in submarine operator

### PR DESCRIPTION
**Pull request by liuxunorg**
### What is this PR for?
Listen to k8s pod, svc creation, destruction and other events in the submarine operator.
* [x] - Submarine CRD auto register
* [x] - Submarine Cluster SVC add/remove event listen
* [x] - Submarine Cluster POD add/remove event listen

#### How To Test
```
cd submarine-cloud
make
./submarine-operator --kubeconfig=$KUBECONFIG --alsologtostderr
```
open new console
```
kubectl apply test1.yaml
kubectl apply test2.yaml
kubectl delete test2.yaml
kubectl delete test2.yaml
```

### What type of PR is it?
Feature

### What is the Jira issue?
* https://issues.apache.org/jira/browse/SUBMARINE-316

### How should this be tested?
* [CI Pass](https://travis-ci.org/liuxunorg/submarine/builds/624659081)

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: submarine-cloud/go.mod [hunk 1]
```diff
@@ -8,10 +8,12 @@ require (
 	github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
 	github.com/mitchellh/go-homedir v1.1.0
 	github.com/spf13/cobra v0.0.5
+	github.com/spf13/pflag v1.0.5
 	github.com/spf13/viper v1.5.0
 	k8s.io/api v0.0.0-20191121015604-11707872ac1c
-	k8s.io/apimachinery v0.0.0-20191121015412-41065c7a8c2a
```
Comment (yuanzac) [inline submarine-cloud/go.mod]: The dependency version is updated.
It's better for us to fix the dependency version.

File: submarine-cloud/pkg/apis/submarine/v1alpha1/default.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ */
+package v1alpha1
+
+import (
+	"github.com/golang/glog"
+	kapiv1 "k8s.io/api/core/v1"
+)
+
+// IsSubmarineClusterDefaulted check if the SubmarineCluster is already defaulted
+func IsSubmarineClusterDefaulted(rc *SubmarineCluster) bool {
```
Comment (yuanzac) [inline submarine-cloud/pkg/apis/submarine/v1alpha1/default.go]: Maybe we can change it to IsDefaultSubmarineCluster

File: submarine-cloud/pkg/apis/submarine/v1alpha1/types.go [hunk 3]
```diff
@@ -14,36 +14,209 @@
...
+
+// ClusterStatus Submarine Cluster status
+type ClusterStatus string
+
+const (
+	// ClusterStatusOK ClusterStatus OK
+	ClusterStatusOK ClusterStatus = "OK"
+
+	// ClusterStatusKO ClusterStatus KO
+	ClusterStatusKO ClusterStatus = "KO"
```
Comment (yuanzac) [inline submarine-cloud/pkg/apis/submarine/v1alpha1/types.go]: I'm not sure about the meanings of KO?

File: submarine-cloud/go.mod (hunk 1 above)
Comment (liuxunorg) [reply submarine-cloud/go.mod]: This is imported automatically and there is no problem with this.

File: submarine-cloud/pkg/apis/submarine/v1alpha1/types.go (hunk 3 above)
Comment (liuxunorg) [reply submarine-cloud/pkg/apis/submarine/v1alpha1/types.go]: Change to `Error`.

File: submarine-cloud/pkg/apis/submarine/v1alpha1/default.go (hunk 2 above)
Comment (liuxunorg) [reply submarine-cloud/pkg/apis/submarine/v1alpha1/default.go]: Done.

**Comment by liuxunorg** [comment]
Will merge if no more comments.

**Pull request closed without merging**